roofit/histfactory/src/HistoToWorkspaceFactory.cxx

```diff
@@ -998,13 +994,11 @@ namespace HistFactory{
     //
     RooRealVar* poi = 0; // (RooRealVar*) POIs->first();
     // for results tables
-    TIterator* params_itr=POIs->createIterator();
-    TObject* params_obj=0;
-    while((params_obj=params_itr->Next())){
+    for (auto const *params_obj : static_range_cast<RooRealVar *>(*POIs)) {
```

Comment:
```suggestion
    for (auto const *poi : static_range_cast<RooRealVar *>(*POIs)) {
```
Look at the line below that can now deleted, you are redundantly casting to RooRealVar.